workers_launch_template.tf

```
@@ -246,230 +140,123 @@ resource "aws_autoscaling_group" "workers_launch_template" {
 }
 
 resource "aws_launch_template" "workers_launch_template" {
-  count = var.create_eks ? (local.worker_group_launch_template_count) : 0
+  for_each = var.create_eks ? local.worker_groups_launch_template_with_defaults : {}
   name_prefix = "${coalescelist(aws_eks_cluster.this[*].name, [""])[0]}-${lookup(
-    var.worker_groups_launch_template[count.index],
+    each.value,
     "name",
-    count.index,
+    each.key,
   )}"
 
   network_interfaces {
-    associate_public_ip_address = lookup(
-      var.worker_groups_launch_template[count.index],
-      "public_ip",
-      local.workers_group_defaults["public_ip"],
-    )
-    delete_on_termination = lookup(
-      var.worker_groups_launch_template[count.index],
-      "eni_delete",
-      local.workers_group_defaults["eni_delete"],
-    )
+    associate_public_ip_address = each.value.public_ip
+    delete_on_termination       = each.value.eni_delete
     security_groups = flatten([
       local.worker_security_group_id,
       var.worker_additional_security_group_ids,
-      lookup(
-        var.worker_groups_launch_template[count.index],
-        "additional_security_group_ids",
-        local.workers_group_defaults["additional_security_group_ids"],
-      ),
+      each.value.additional_security_group_ids,
     ])
   }
 
   iam_instance_profile {
-    name = coalescelist(
-      aws_iam_instance_profile.workers_launch_template.*.name,
-      data.aws_iam_instance_profile.custom_worker_group_launch_template_iam_instance_profile.*.name,
-    )[count.index]
+    name = coalesce(
+      aws_iam_instance_profile.workers_launch_template[each.key].name,
+      data.aws_iam_instance_profile.custom_worker_group_launch_template_iam_instance_profile[each.key].name,
+    )
   }
 
   enclave_options {
-    enabled = lookup(
-      var.worker_groups_launch_template[count.index],
-      "enclave_support",
-      local.workers_group_defaults["enclave_support"],
-    )
+    enabled = var.worker_groups_launch_template[each.key].enclave_support
```

This fails. did you mean:
```suggestion
    enabled = each.value.enclave_support
```

---

changed :heavy_check_mark: